scripts/visual-active-effects.mjs

```diff
@@ -36,17 +36,46 @@ export class VisualActiveEffects extends Application {
       // Set up the various text (intro and content).
       const desc = entry.effect.flags["dfreds-convenient-effects"]?.description;
       const data = entry.effect.flags[MODULE]?.data ?? {};
+      
+      // Get the effect rollData to populate enrichers within the descriptions.
+      let rollData = {};
+      try {
+        if (entry.effect.origin) {
+          let origin = await fromUuid(entry.effect.origin);
+          if (origin?.documentName === ActiveEffect.documentName) {
+            // Change the origin to the parent of the ActiveEffect - the originating item or actor.
+            origin = origin.parent;
+          }
+          if (origin?.pack) {
+            // Origin references a compendium, change the origin to the effect's parent - the local item or actor.
+            origin = entry.effect.parent;
+          }
+          if (typeof origin?.getRollData === "function") {
+            rollData = origin.getRollData();
+          }
+        }
+
+        // Fallback to the parent if there's no rollData.
+        if (!rollData || !Object.keys(rollData).length) {
+          if (typeof entry.effect.parent?.getRollData === "function") {
+            rollData = entry.effect.parent.getRollData();
+          }
+        }
```

Comment:
```suggestion
        if (!rollData) rollData = entry.effect.parent.getRollData();
```

A bit unneeded since effects always live on an item or actor, which always have a `getRollData` method, so this can act as the fallback.